articles/introduction-to-r/index.md

```diff
@@ -0,0 +1,328 @@
+```R
+vector_num <- c(2,4,6,8)
+vector_char <- c('a','b','c','c')
+vecor_logic <- c(FALSE, TRUE)
+```
+All operations such as addition, subtraction, multiplication, and division can be performed on vectors. Do give it a try. For your reference, I have attached the code below. Try it out on the console and observe the output.
+
+```r
+a <- c(1,2,3,4) # vector definition
+b <- c(2,4,6,8)  # vector definition
+a + b # addition
+a - b # subtraction
+a * b # multiplication
+a/ b # division
+a %% b # modulus(returns the remainder after division)
+```
+##### Naming Indices
+
+R lets us identify each index with a custom name. For example, let us define a vector `first_quarter` and initialize it with numbers 1,2, and 3. We represent another vector, `first_quarter_names`, with names of the first three months.
+
+Using the `names` function, we can assign unique indexes to elements of `first_quarter.` This helps in customizing the index names, which is useful in cases dealing with large amounts of data.
+
+```r
+first_quarter <- c(1,2,3)
+first_quarter_names <- c("Jan","Feb", "Mar")
+names(first_quarter) <- first_quarter_names
+```
+The `names` function helps us label and tabulate the data for better understanding in the future — an essential tool for all data analysts across the world. Using the new indices created, we can replace them in place of the default array indices. Let us consider the example below, which illustrates the same.
+
+```r
+first_quarter["Jan"]
+```
+
+The above piece of code returns 1. Hence, we can use the above pair of lists and names as key-value pairs. Key-value pairs are beneficial in implementing [hash tables](https://www.section.io/engineering-education/bloom-filters-data-structure/), storing information, etc.
+
+##### Slicing Lists
+
+Lists can also be sliced. Slicing refers to the operation of obtaining a subset of serial elements from the list. For example, to get the first two months mentioned in the list `first_quarter,` we can use the following code.
+```r
+list_slice <- first_quarter[c(1:2)]
+```
+#### Matrices
+
+Matrices are 2-dimensional (rectangular) structures used commonly in the field of computer science. One application of matrices is in the field of computer vision. Each image is treated as a stack of three matrices, each representing a color channel. R offers libraries highly optimized for matrix operations. Let us look at defining matrices.
+
+![three channels in an image](image_channel.jpg)
+
+*[Image Source](https://www.sketchpad.net/channels1.htm)*
+
+##### Defining a Matrix
+
+In R, we create new matrices using the `matrix()` function. The `matrix()` function takes a vector containing the elements, the orientation, and the number of rows. The orientation refers to the order of elements being stored. When the argument `byrow` is `TRUE,` the elements are stored row-wise. To store column-wise, we define `bycol` to be `TRUE.`
+
+```r
+matrix(1:16, bycol=TRUE, ncol=4)
```

Comment:
Add a comment explaining the dimensions of the matrix it will yield, and explaining what dimension corresponds to the column-wise definition.